docs/rfcs/1005-realm-key-rotation.md

```diff
@@ -0,0 +1,803 @@
+            //
+            "name": "timestamp",
+            "type": "DateTime"
+        },
+        {
+            "name": "realm",
+            "type": "VlobID"
+        },
+        {
+            "name": "keys_bundle_id",
+            // Vlob within the realm, the version is always 1
+            "type": "VlobID"
+        },
+        {
+            "name": "keys_bundle_key",
+            "type": "SecretKey"
+        }
+    ]
+}
+
+```
+
+### Existing manifest model change: `user_manifest`
+
+```json5
+{
+    "label": "UserManifest",
+    "type": "user_manifest",
+    "other_fields": [
+        // [...] ignored elements
+        {
+            "name": "workspaces",
+            "type": "List<WorkspaceEntry>"  // <-- change the WorkspaceEntry type
+        }
+    ]
+}
+```
+
+`WorkspaceEntry` change:
+
+```json5
+{
+    "label": "WorkspaceEntry",
+    "fields": [
+        {
+            "name": "id",
+            "type": "VlobID"
+        },
+        {
+            "name": "name",
+            "type": "EntryName"
+        },
+        {
+            "name": "keys_location",
+            "type": "VlobID"
+        }
+        // Removed fields:
+        // - key
+        // - encryption_revision
+        // - encrypted_on
+        // - role_cached_on
+        // - role_cache_value
+    ]
+}
+```
+
+> **Note:**
+>
+> `role_cached_on` and `role_cache_value` are not directly related to this RFC, but
+> given they are no longer in use in Parsec v3, we also remove them.
+
+The field `keys_locations` contains a `VlobID` that point on a vlob belonging to the user realm.
+This vlob contains a `RealmKeysManifest`:
+
+```json5
+{
+    "label": "RealmKeysManifest",
+    "type": "realm_keys_manifest",
+    "other_fields": [
+        {
+            "name": "realm_id",
+            "type": "VlobID"
+        },
+        {
+            "name": "keys",
+            // Ordered by index
+            "type": "List<SecretKey>"
+        }
+    ]
+}
+```
+
+TODO: keys bundle system:
+
+- on key-rotation, owner upload a new vlob containing all the decryption keys (including
+  the new one), and itself encrypted by a symmetric key. This is the keys bundle.
+- the symmetric key is then send by message to all the members of the workspace.
+- each member update it user manifest with the vlob id and key of the keys bundle.
+
+Given the keys bundle is stored as a vlob, anyone with write access can upload new versions
+of it, so the client always only retrieve the version 1 of the vlob.
+
+Three benefits for this approach:
+
+- Keys are encrypted a single time instead of for each member: faster to do and less
+  disk space consumed (it must be stored for each user as message and in user manifest)
+- When access is removed, the user is no longer able to get back the decryption keys
+  (instead of having them in it user manifest). The client can of course save them when
+  it has access to them, but it requires a modified Parsec.
```

Comment:
In that case the user would only be able to read the data that where using the previous decryption keys ? Without a modified parsec, the user could still copy the data on another support.

Reply:
> In that case the user would only be able to read the data that where using the previous decryption keys ?

Yes, no matter what new data are not accessible to him

> Without a modified parsec, the user could still copy the data on another support.

Yes totally, hence no having the key stored on the local machine is only a hardening element